app/controllers/pension_types_controller.rb

```diff
@@ -0,0 +1,8 @@
+class PensionTypesController < ApplicationController
+  def next
+    question = GuideDecorator.cached_for(GuideRepository.new.find(params[:question_id]))
+    answer = question.metadata.answers[params[:response]]
+
+    redirect_to answer || params[:question_id]
```

Comment:
Are values for `question_id` whitelisted / checked anywhere? This could be an open redirect / forwarding vector: https://www.owasp.org/index.php/Top_10_2013-A10-Unvalidated_Redirects_and_Forwards